sw/device/silicon_creator/lib/ownership/owner_verify.c

```diff
@@ -0,0 +1,151 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/ownership/owner_verify.h"
+
+#include "sw/device/silicon_creator/lib/base/util.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/sigverify/ecdsa_p256_key.h"
+#include "sw/device/silicon_creator/lib/sigverify/sigverify.h"
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/verify.h"
+
+// A version of spx_verify that is tailored to ROM_EXT use cases.
+// In particular:
+//   - We don't care about the OTP setting for SPX+ in the ROM_EXT.
+//   - We don't care about flash_exec in the ROM_EXT.
```

Comment:
As discussed offline, lets think about adding a magic constant so that we can add redundant checks. This can be done in a follow up PR.

Reply:
I'd like to take care of the magic constant for redundancy in a stand-alone PR so that it can be reviewed in isolation (without a bunch of other code moving and refactoring).